department-of-veterans-affairs/abd-vro
# MCP-2048 Cleanup Python Dockerfiles

MCP-2048 Cleanup Python Dockerfiles

## What was the problem?

Each python microservice has a Dockerfile that is the same as the others except for the folder that is being copied. We'd need to create one for each diagnostic code which means future docker updates would have to happen to every file

Also cleaning up the `examples` folder in `pdfgenerator` since its not useful anymore

Associated tickets or Slack threads:

- [MCP-2048](https://amida.atlassian.net/browse/MCP-2048)

## How does this fix it?[^1]

Remove all the Python Dockerfiles and have 1 version in `service-python`

This new Dockerfile uses build stages for each python service and then a final stage that selects the appropriate build stage based on the `SERVICE_FOLDER` build arg. If no argument is provided, it defaults to `layer-gradle` which will copy over the appropriate files for the build to succeed. On the other hand, running a command like `docker build --build-arg SERVICE_FOLDER=pdfgenerator -f Dockerfile .` will select `layer-pdfgenerator` which will copy over selected service.

## How to test this PR
- Navigate to `service-python` and run `docker build --build-arg SERVICE_FOLDER=pdfgenerator -f Dockerfile .` to test the local build process. Feel free to replace `SERVICE_FOLDER` with any other python folder name within `service-python`
- Run `./gradlew build check docker` to test the normal process
- Step 2[^secrel]


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
